nixos/modules/services/web-apps/pretalx.nix

```diff
@@ -0,0 +1,406 @@
+    # https://docs.pretalx.org/administrator/installation.html
+
+    services.postgresql = mkIf (cfg.database.createLocally && cfg.settings.database.backend == "postgresql") {
+      ensureUsers = [ {
+        name = cfg.settings.database.user;
+        ensurePermissions = {
+          "DATABASE ${cfg.settings.database.name}" = "ALL PRIVILEGES"; };
+      } ];
+      ensureDatabases = [ cfg.settings.database.name ];
+    };
+
+    services.redis.servers.pretalx.enable = true;
+
+    services.nginx = mkIf cfg.nginx.enable {
+      additionalModules = with pkgs.nginxModules; [
+        moreheaders
+      ];
+      recommendedGzipSettings = mkDefault true;
+      recommendedOptimisation = mkDefault true;
+      recommendedProxySettings = mkDefault true;
+      recommendedTlsSettings = mkDefault true;
+      virtualHosts.${cfg.nginx.domain} = {
+        # https://docs.pretalx.org/administrator/installation.html#step-7-ssl
+        extraConfig = ''
+          gzip off;
+          more_set_headers Referrer-Policy same-origin;
+          more_set_headers X-Content-Type-Options nosniff;
+        '';
+        locations."/" = {
+          proxyPass = "http://unix:/run/pretalx/pretalx.sock";
+        };
+        locations."/media/" = {
+          alias = "/var/lib/pretalx/data/media/";
+          extraConfig = ''
+            access_log off;
+            more_set_headers Content-Disposition 'attachment; filename="$1"';
+            expires 7d;
+            gzip on;
+          '';
+        };
+        locations."/static/" = {
+          alias = cfg.settings.filesystem.static;
+          extraConfig = ''
+            access_log off;
+            more_set_headers Cache-Control "public";
+            expires 365d;
+            gzip on;
+          '';
+        };
+      };
+    };
+
+    systemd.services = let
+      commonUnitConfig = {
+        environment = {
+          inherit PYTHONPATH;
+          PRETALX_CONFIG_FILE = format.generate "pretalx.cfg" cfg.settings;
+        };
+        serviceConfig = {
+          DynamicUser = true;
+          User = "pretalx";
+          Group = "pretalx";
+          StateDirectory = "pretalx";
+          LogsDirectory = "pretalx";
+          WorkingDirectory = "/var/lib/pretalx";
+          RuntimeDirectory = "pretalx";
+          SupplementaryGroups = [
+            "redis-pretalx"
+          ];
+        };
+      };
+    in {
+      pretalx-web = recursiveUpdate {
+        description = "pretalx web service";
+        after = [
+          "network.target"
+          "redis-pretalx.service"
+        ] ++ lib.optionals (cfg.settings.database.backend == "postgresql") [
+          "postgresql.service"
+        ] ++ lib.optionals (cfg.settings.database.backend == "mysql") [
+          "mysql.service"
+        ];
+        wantedBy = [
+          "multi-user.target"
+        ];
+        preStart = ''
+          mkdir -p $STATE_DIRECTORY/media
+          ln -sf ${cfg.package}/bin/pretalx-manage $STATE_DIRECTORY/pretalx-manage
```

Comment:
This should probably be exposed as a `environment.systemPackages`-wide script to do admin of the pretalx